subsys/net/lib/aws_fota/src/aws_fota_json.c

```diff
@@ -115,13 +118,34 @@ int aws_fota_parse_DescribeJobExecution_rsp(const char *job_document,
 
 	cJSON *hostname = cJSON_GetObjectItemCaseSensitive(location, "host");
 	cJSON *path = cJSON_GetObjectItemCaseSensitive(location, "path");
+	cJSON *url = cJSON_GetObjectItemCaseSensitive(location, "url");
 
-	if ((cJSON_GetStringValue(hostname) != NULL)
+	if (cJSON_GetStringValue(url) != NULL) {
+		struct http_parser_url u;
+
+		http_parser_url_init(&u);
+		http_parser_parse_url(url->valuestring,
+				strlen(url->valuestring), false, &u);
+
+		uint16_t parsed_host_len =
+			MIN(max_hostname_len, u.field_data[UF_HOST].len);
+		uint16_t parsed_file_len =
```

Comment:
We need to handle the scenario where the passed in `max_hostname_len` and/or `max_file_path_len` are too small. I.E smaller than the lengths returned from `http_parser_url`.

If this occurs there is no error handling. `parsed_host_len` too small and the file and hostname bufs will be cut without containing the complete strings.